cloudferry/lib/os/storage/plugins/copy_mechanisms.py

```diff
@@ -82,27 +83,47 @@ def copy(self, context, source_object, destination_object):
         src_host = source_object.host
         dst_host = destination_object.host
 
-        rr = remote_runner.RemoteRunner(src_host, src_user)
+        rr_src = remote_runner.RemoteRunner(src_host, src_user, sudo=True)
+        rr_dst = remote_runner.RemoteRunner(dst_host, dst_user, sudo=True)
 
         ssh_opts = ('-o UserKnownHostsFile=/dev/null '
                     '-o StrictHostKeyChecking=no')
 
+        # Choose auxiliary port for SSH tunnel
+        aux_port_start, aux_port_end = \
+            context.cfg.migrate.storage_backend_timeout.split('-')
+        aux_port = random.randint(aux_port_start, aux_port_end)
+
         try:
             progress_view = ""
-            if files.is_installed(rr, "pv"):
-                src_file_size = files.remote_file_size(rr, source_object.path)
+            if files.is_installed(rr_src, "pv"):
+                src_file_size = files.remote_file_size(rr_src,
+                                                       source_object.path)
                 progress_view = "pv --size {size} --progress | ".format(
                     size=src_file_size)
 
-            copy = ("dd if={src_file} | {progress_view} "
-                    "ssh {ssh_opts} {dst_user}@{dst_host} "
-                    "'dd of={dst_device}'")
-            rr.run(copy.format(src_file=source_object.path,
-                               dst_user=dst_user,
-                               dst_host=dst_host,
-                               ssh_opts=ssh_opts,
-                               dst_device=destination_object.path,
-                               progress_view=progress_view))
+            # First step: prepare netcat listening on aux_port on dst and
+            # forwarding all the data to block device
+            dump_cmd = ('/bin/bash -c \'nc -l -p {aux_port} -s 127.0.0.1 | '
+                        'dd of={dst_device}\' &')
```

Comment:
Am I reading this line correctly: `nc` will be spawned for each migrated volume in the background and will never be killed? The `&` sign in the end confuses me, and also I don't see any termination for the spawned `nc` process. If I'm reading it correctly, this needs to be resolved, we should not leave `nc` processes running in the background.